### PR TITLE
[-] FO : Checkout - displays more details

### PR DESCRIPTION
**Pull request by Shine-neko**
## Forge Ticket

http://forge.prestashop.com/browse/BOOM-413

### CODE REVIEW

File: themes/classic/templates/checkout/_partials/cart-voucher.tpl [hunk 1]
```diff
@@ -3,7 +3,8 @@
     <div class="cart-voucher">
       <ul class="promo-name">
         {foreach from=$cart.vouchers.added item=voucher}
-          <li>{$voucher.name} <a class="pull-xs-right" href="{$voucher.delete_url}" data-link-action="remove-voucher">{l s='Cancel'}</a></li>
+          <li>{$voucher.name} {if $voucher.reduction_amount != '0.00'}{$voucher.reduction_amount_formatted} {/if} {if $voucher.reduction_percent != '0.00'} {$voucher.reduction_percent}%{/if}
```
Comment (maximebiloe) [inline themes/classic/templates/checkout/_partials/cart-voucher.tpl]: The .tpl file does not have to calculate if it's a percentage or an amount, the presenter should give you the formatted string. Could you change that, please ?
Also, of the value is 0, it should not be displayed, so the presenter should not send the voucher to the .tpl

File: src/Adapter/Cart/CartPresenter.php [hunk 2]
```diff
@@ -295,8 +292,13 @@ private function getTemplateVarVouchers(Cart $cart)
         $vouchers = [];
 
         foreach ($cartVouchers as $cartVoucher) {
+
```
Comment (maximebiloe) [inline src/Adapter/Cart/CartPresenter.php]: Can you remove the unnecessary line, please ?

File: themes/classic/templates/checkout/_partials/cart-voucher.tpl (hunk 1 above)
Comment (Shine-neko) [inline themes/classic/templates/checkout/_partials/cart-voucher.tpl]: Yep

File: src/Adapter/Cart/CartPresenter.php [hunk 3]
```diff
@@ -16,6 +16,7 @@
 use Configuration;
 use CartRule;
 use Tools;
+use Symfony\Component\VarDumper\VarDumper;
```
Comment (maximebiloe) [inline src/Adapter/Cart/CartPresenter.php]: I think you used it for debug, but can you remove it, please?

File: themes/classic/templates/checkout/_partials/cart-voucher.tpl [hunk 4]
```diff
@@ -3,7 +3,11 @@
     <div class="cart-voucher">
       <ul class="promo-name">
         {foreach from=$cart.vouchers.added item=voucher}
-          <li>{$voucher.name} <a class="pull-xs-right" href="{$voucher.delete_url}" data-link-action="remove-voucher">{l s='Cancel'}</a></li>
+          <li>{$voucher.name}
+            <div class="pull-xs-right">
+              {$voucher.reduction_formated}
+              <a class="pull-xs-right" href="{$voucher.delete_url}" data-link-action="remove-voucher"><i class="material-icons">{l s='delete'}</i></a></li>
```
Comment (maximebiloe) [inline themes/classic/templates/checkout/_partials/cart-voucher.tpl]: Are you sure this is the right location? :stuck_out_tongue:

Comment (Shine-neko) [inline themes/classic/templates/checkout/_partials/cart-voucher.tpl]: Gneuh ?

Comment (maximebiloe) [inline themes/classic/templates/checkout/_partials/cart-voucher.tpl]: The remove button should be at right of the voucher name. If we let it here, the amounts are not aligned.

**Comment by Shine-neko** [comment]
Youhou @maximebiloe

**Comment by maximebiloe** [comment]
Thank you.
Can you redo it on `StarterTheme`, please?

**Pull request merged**